File: app/SmartHome/app/src/main/java/com/example/josephpham/smarthome/model/Mode.kt
package com.example.josephpham.smarthome.model

import java.time.LocalDateTime
import java.util.*

/**
 * Created by Joseph Pham on 1/8/2018.
 */
class Mode {
    var id:Int = 0
    var mode_name: String =""
    var id_user: Int =  0
    var status: Int = 0
    var start_time : Date? = null
    var end_time : Date? = null
    constructor(id: Int, mode_name: String, id_user: Int, status: Int){
        this.id = id
        this.mode_name = mode_name
        this.id_user = id_user
        this.status = status
    }
    constructor(id: Int, mode_name: String, id_user: Int, status: Int, start_time: Date, end_time: Date){
        this.id = id
        this.mode_name = mode_name
        this.id_user = id_user
        this.status = status
        this.start_time = start_time
        this.end_time = end_time
    }
}

File: Server/model/mode.js
'use strict'
var mongoose = require('mongoose');
var conn = require('../util/config').database;

var schemaMode = new mongoose.Schema({
	mode_name : {type : String},
	id_user : {type : mongoose.Schema.Types.ObjectId, ref : 'User'},
	status : {type : Boolean, default : true},
	
//	type: {type : Number},//0: normal: thuc hien cac ngay trong tuan hoac hang ngay; 1: special: thực hiện trong 1 khoảng thời gian
	
	circle : {type : []},
	starttime : {type : Number},
	stoptime : {type : Number}
	
//	day : {startdate : {type : Date}, stopdate : {type : Date}}
}, { toJSON : {virtuals: true}});

schemaMode.virtual('modedetail', {ref : 'ModeDetail', localField : '_id', foreignField : 'mode', justOne : false});

var Mode = conn.model('Mode', schemaMode, 'MODE');

module.exports = exports = Mode;
